common/http/src/client.rs

```diff
@@ -89,7 +82,7 @@ impl Client {
         self.last_retry.elapsed() > Duration::from_secs(3)
     }
 
-    fn flush(&mut self) {
+    async fn flush(&mut self) {
```

Comment:
Because flush takes a &mut self it's impossible for multiple threads to call flush at the same time as the borrow checker disallows the creation of multiple mutable references. With flush being private it's also impossible for an external caller to interleave send and flush incorrectly, so I think the type system eliminates the race condition here

Reply:
oh true!
that's a nice guarantee by compiler!